src/main/java/Parser.java

```diff
@@ -0,0 +1,78 @@
+public class Parser {
+    public static Task parseTask(String type, String input) {
+        int descriptionIndex;
+        String description;
+        Task newTask;
+
+        switch (type) {
+        case InputManager.TODO_PHRASE:
+            descriptionIndex = input.indexOf(InputManager.TODO_PHRASE);
+
+            // add one to remove space
+            description = input.substring(descriptionIndex + InputManager.TODO_PHRASE.length() + 1,
+                    input.length());
+
+            newTask = new Todo(description);
+            break;
+        case InputManager.DEADLINE_PHRASE:
+            descriptionIndex = input.indexOf(InputManager.DEADLINE_PHRASE);
+            int byIndex = input.indexOf(InputManager.BY_PHRASE);
+
+            // add one to remove space
+            // minus one to remove space
+            description = input.substring(
+                    descriptionIndex + InputManager.DEADLINE_PHRASE.length() + 1, byIndex - 1);
+            String by =
+                    input.substring(byIndex + InputManager.BY_PHRASE.length() + 1, input.length());
+
+            newTask = new Deadline(description, by);
+            break;
+        case InputManager.EVENT_PHRASE:
+            descriptionIndex = input.indexOf(InputManager.EVENT_PHRASE);
+            int atIndex = input.indexOf(InputManager.AT_PHRASE);
+
+            // add one to remove space
+            // minus one to remove space
+            description = input.substring(descriptionIndex + InputManager.EVENT_PHRASE.length() + 1,
+                    atIndex - 1);
+            String at =
+                    input.substring(atIndex + InputManager.AT_PHRASE.length() + 1, input.length());
+
+            newTask = new Event(description, at);
+            break;
+
```

Comment:
You could remove the newline to keep the overall code similar.

Reply:
Thanks for pointing this out! It has been addressed